core/src/mode.rs

```diff
@@ -1,14 +1,17 @@
 //! Control in the different modes by which a source file can be parsed.
 
 /// The mode argument specifies in what way code must be parsed.
-#[derive(Clone, Copy, Hash, PartialEq, Eq)]
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq)]
 pub enum Mode {
     /// The code consists of a sequence of statements.
     Module,
     /// The code consists of a sequence of interactive statement.
     Interactive,
     /// The code consists of a single expression.
     Expression,
+    /// The code consists of a sequence of statements which are part of a
+    /// Jupyter notebook and thus could include magics.
```

Comment:
Nit: Add a link to what you mean by *magics*. It otherwise sounds very wizardly for someone unfamiliar to what magics. are

Reply:
Can you give a quick read at the updated documentation for `Mode::Jupyter` and see if it covers everything?